chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt

```diff
@@ -170,6 +172,48 @@ class RoomSpek : Spek({
 
             assertThat(badEvents).isEmpty()
         }
+
+        it("always has the correct memberUserIds after room data is changed") {
+
+            setUpInstanceWith(createDefaultRole(), newUsers(PUSHERINO, ALICE), newRoom(GENERAL, ALICE))
+            val superUser = chatFor(SUPER_USER).connect().assumeSuccess()
+            val roomUpdated = CountDownLatch(1)
+
+            superUser.subscribeToRoomMultipart(superUser.generalRoom) { event ->
+                when (event) {
+                    is RoomEvent.RoomUpdated -> {
+                        roomUpdated.countDown()
+                    }
+                }
+            }
+
+            assertThat(superUser.generalRoom.memberUserIds.size).isEqualTo(2) //alice and super user
+            superUser.updateRoom(superUser.generalRoom, customData = mapOf("key" to "data") ).assumeSuccess()
+            roomUpdated.await()
+            assertThat(superUser.generalRoom.memberUserIds.size).isEqualTo(2)
+
+        }
+
+        it("always has the correct memberUserIds after message is sent to a room") {
+
+            setUpInstanceWith(createDefaultRole(), newUsers(PUSHERINO, ALICE), newRoom(GENERAL, ALICE))
+            val superUser = chatFor(SUPER_USER).connect().assumeSuccess()
+            val roomUpdated = CountDownLatch(1)
+
+            superUser.subscribeToRoomMultipart(superUser.generalRoom) { event ->
+                when (event) {
+                    is RoomEvent.MultipartMessage -> {
```

Comment:
This should also generate a RoomUpdated event (for the change in unreadCount) which is what I think we should wait for here, because we know that is the code path which previously had the bug.
The ordering between the MultipartMessage and RoomUpdated events is not specified, so we may or may not have processed the room update when this event comes through.

Reply:
That makes sense - I've incremented the count `CountDownLatch` to 2, and decreased it in the `RoomEvent.RoomUpdated` callback so that both have to be called before we do our assertions. Does that make sense?

Reply:
That seems reasonable, though I'm not sure that waiting for `MultipartMessage` actually serves a purpose.

Reply:
Okay. What would you prefer? I can keep the method callback and decrementing the `CountDownLatch` or I can remove that block of code?

Reply:
I'd say if we don't have a reason to wait on `MultipartMessage`, then we shouldn't.

Reply:
Done.